### DISCUSSION TITLE
Live grep navigation

### DISCUSSION THREAD

**Discussion by martin-braun** [vscode-neovim/vscode-neovim#1296] [Ideas]
Replicating Live Grep from Telescope as close as possible: 

![2023-07-06 17 03 54](https://github.com/vscode-neovim/vscode-neovim/assets/24259245/d3da8b3c-69a5-41a9-89a0-3d4e8fd3ed55)

I mapped `<C-/>` to enter the full text search, what is the default for `vscode-neovim` or is there none already?

The I can navigate the search results with `<C-n>` and `<C-p>`. The search will be erased when I hit `<C-c>` while being in the search result / search viewlet or when I am in normal mode in the text editor. So hitting `<C-c>` twice will always erase the search.

I did it this way, because `<C-n>` opens a new buffer and I also use `<C-p>` to open files by filename (go to file ...). So when I'm done and want to open a new buffer / another file I just hit `<C-c>` twice.

The search results can also be navigated with arrow keys, but those won't focus the text editor automatically.

Does this behavior make sense as defaults for `vscode-neovim`? Thoughts?

**Comment by theol0403** [comment]
The default is C-S-F, a built-in vscode binding. 

Nice bindings! Would be good to include probably. Note that you can probably add a when clause to the big one in `escape` and `ctrl+c` that will send the escape to vscode before it sends it to nvim.

**Reply by martin-braun** [reply]
@theol0403 I doubt I can do that. I have to let Neovim handle the `ctrl+c` in normal mode exclusively, because I have to fire two `VSCodeCall`s:

```<CMD>call VSCodeCall('search.action.clearSearchResults')<CR><CMD>call VSCodeCall('workbench.action.focusActiveEditorGroup')<CR>```

It seems `search.action.clearSearchResults` will focus the search viewlet and afaik VSCode cannot have multiple actions on a binding.

So long story short, we have to rebind `<C-c>` in normal mode to make it work.

(Btw, this also fixed the strange behavior that Neovim would tell me to type `:qa` to exit Neovim)

**Reply by theol0403** [reply]
I think you can actually! https://code.visualstudio.com/docs/getstarted/keybindings#_running-multiple-commands

**Reply by martin-braun** [reply]
@theol0403 Nice! So your recommendation is to catch `<C-c>` in normal mode by VSCode, instead of catching it in Neovim and sending it to VSCode via `VSCodeCall`, right?

**Comment by martin-braun** [comment]
@theol0403 I try to do the same for the normal find search, but I fail to find the correct command in VSCode. I basically want to map `<C-c>` and `<ESC>` to be identical in normal mode when it comes to search cancelation on all levels.

Could you please advice?

**Reply by theol0403** [reply]
https://github.com/vscode-neovim/vscode-neovim/blob/9479c650a3ff79382c6b86e84b16bc317fd3894f/package.json#L372

Maybe you can add a when clause that sends escape to vscode instead so that it cancels your search. I'm not sure how your search cancellation works, however.

**Reply by martin-braun** [reply]
@theol0403 It's been a while, I forgot a bit about this discussion, since I had a lot of things to focus on. Been using my mappings for quite a while without addressing the difference between `<C-c>` and `<ESC>` and I have to say, it's better they don't collide. 

Sometimes I would search globally and also search within my results locally. The difference would give me better control. I could cancel the in-file search using `<ESC>` without canceling the global search and vice-versa.

Did anything in regards of my idea got implemented that I didn't notice? Otherwise I might turn my idea to a PR in some time, as I don't want my idea to become stale in the very end.

Best regards.
